ls1intum/Artemis
# `General`: Let exercise and lecture list rows behave as links

### Checklist
#### General


#### Client

### Motivation and Context



Exercise and lecture list rows (student perspective) do not behave as link, but using JS navigate calls instead.
This PR let's them behave as links so you can open them in new tabs with a middle mouse click, for example.
Closes #2820.

### Description


Utilizing Boostraps stretched link class to overlay the existing rows with a real link tag.

### Steps for Testing

Prerequisites:
- 1 Student
- A course with exercises and lectures

1. Log in to Artemis
2. Navigate to Exercises
3. Verify that exercise row do have a real link on them (e.g. open in new tab to verify, use middle mouse click)
4. Verify that buttons on these rows still work as before and are *not* overlayed by the stretched link
5. Do the same in the lecture list

### Review Progress





#### Code Review
#### Manual Tests

### Screenshots




https://user-images.githubusercontent.com/23171488/144301127-e66ed740-2359-45a8-acb5-de8ff2a0d30c.mov
